docs/application/native/guides/device/battery-monitor.md

```diff
@@ -79,90 +77,105 @@ To get the battery usage information related to the application ID:
+        printf("The Battery Usage for appid [%s] is not recorded");
     else
         printf("Error Occurred [%d]", error_code);
 
     ```
 
-- Get the battery usage information of an application ID for each resource over a specific duration of time.
-  To get the resource ID, use the `battery_monitor_resource_id_e` enum and to get the time duration type information, use the `battery_monitor_duration_type_e` enum:
+- Get the battery usage information of an application ID for each resource over a specific interval of time.
+  To get the resource ID, use the `battery_monitor_resource_id_e` enum. The timestamps **from** and **to** must be set as per the Unix Epoch time format:
 
     ```
-    battery_monitor_h data_handle = NULL;
-
-
-    /*First use this API to fetch the usage for all the available resources in the data handle*/
-    error_code = battery_monitor_get_usage_by_app_id_for_all_resource_id(app_id, duration_val, &data_handle);
+    battery_usage_data_h data_handle = NULL;
+    /* To get the usage for past thousand seconds from the current epoch time */
+    int period_t = 1000;
+    to = now; from = to - period_t;
 
+    /*Use this API to fetch the usage of all the available resources in the data handle*/
+    error_code = battery_monitor_get_power_usage_by_app_for_all_resources(app_id, from, to, &data_handle);
     if (error_code == BATTERY_MONITOR_ERROR_NONE)
         printf("Data Handle information received");
+    else if (error_code == BATTERY_MONITOR_ERROR_RECORD_NOT_FOUND)
+        printf("The Battery Usage for appid [%s] is not recorded");
     else
         printf("Error Occurred [%d]", error_code);
 
-    /*Now retrieving the information one by one from the data_handle*/
-    battery_monitor_resource_id_e resource_id = BATTERY_MONITOR_RESOURCE_ID_DISPLAY;
+    /*Retrieving the information one by one from the data_handle*/
+    battery_monitor_resource_id_e resource_id = BATTERY_MONITOR_RESOURCE_ID_CPU;
 
-    error_code = battery_monitor_get_usage_for_resource_id(data_handle, resource_id, &battery_usage);
+    error_code = battery_monitor_usage_data_get_power_usage_per_resource(data_handle, resource_id, &battery_usage);
     if (error_code == BATTERY_MONITOR_ERROR_NONE)
-        printf("Battery Usage by display is [%d]", battery_usage);
+        printf("Battery Usage by CPU is [%lf]", battery_usage);
 
     resource_id = BATTERY_MONITOR_RESOURCE_ID_WIFI;
 
-    error_code = battery_monitor_get_usage_for_resource_id(data_handle, resource_id, &battery_usage)
+    error_code = battery_monitor_usage_data_get_power_usage_per_resource(data_handle, resource_id, &battery_usage);
     if (error_code == BATTERY_MONITOR_ERROR_NONE)
-        printf("Battery Usage by display is [%d]", battery_usage);
+        printf("Battery Usage by wifi is [%lf]", battery_usage);
+    else if (error_code == BATTERY_MONITOR_ERROR_RECORD_NOT_FOUND)
+        printf("Battery usage for above resource is not recorded");
```

Comment:
@nawab-samsung Question. According to your other comments, isn't "by" better than "for"?

Battery usage _by_ above resource ...

Reply:
@mijin85cho  I think appid is a value(a reference to an app/ index of an app),  but not the app itself, so it( the value itself) may not consume battery that why for is appropriate. My understanding can be wrong, so kindly give your valuable suggestion, will go as you say.

I do agree that "by" gives better clarity but it is situation-based, 
Also,you had commented with your suggestion https://github.com/Samsung/tizen-docs/pull/1076#discussion_r409361830. 
So kept it intact. 
@d1-agarwal What are your thoughts on this?

Reply:
@d1-agarwal  kindly change it slightly yo: 

```suggestion 
         printf("The Battery Usage for the resource is not recorded");
Refer to the discussion: 
https://github.com/Samsung/tizen-docs/pull/1076/files#r414306294
https://github.com/Samsung/tizen-docs/pull/1076/files#r414417039